File: Chatbot-Studi.com/PublicSiteChatbot.Front/Studi.AI.Chatbot.Front/Client/UserQueryAskingRequestModel.cs
namespace Studi.AI.Chatbot.Front.Client;

using System.Text.Json.Serialization;

public class UserQueryAskingRequestModel
{
    [JsonPropertyName("conversation_id")]
    public Guid ConversationId { get; set; }

    [JsonPropertyName("user_query_content")]
    public string UserQueryContent { get; set; } = "";

    [JsonPropertyName("display_waiting_message")]
    public bool DisplayWaitingMessage { get; set; } = true;

}

File: Chatbot-Studi.com/Chatbot.BackOffice.Front/client_models/user_query_asking_request_model.py
from pydantic import BaseModel
from typing import Union
from uuid import UUID

class UserQueryAskingRequestModel(BaseModel):
    conversation_id: UUID
    user_query_content: str = ""
    display_waiting_message: bool = False

    def to_json(self) -> dict[str, Union[str, UUID]]:
        return {"conversation_id": str(self.conversation_id), "user_query_content": self.user_query_content, "display_waiting_message": self.display_waiting_message}